src/app/code/community/Zendesk/Zendesk/Block/Adminhtml/Dashboard/Grids.php

```diff
@@ -65,15 +65,20 @@ protected function _prepareLayout() {
             } catch (Exception $ex) {
                 $allTicketView = array();
             }
-                
+
+            // Loop through each view ID as per config
             foreach($viewsIds as $viewId) {
-                $view = array_shift(array_filter($allTicketView, function($view) use($viewId) {
-                    return $view['id'] === (int) $viewId;
-                }));
-                
-                $count = array_shift(array_filter($ticketsCounts['view_counts'], function($view) use($viewId) {
+                // Searches for the view's details by matching all views retrieved from the api to the current view id
+                $view = array_filter($allTicketView, function($ticketView) use($viewId) {
+                    return $ticketView['id'] === (int) $viewId;
+                });
+                // Return only the first value (usually returns just 1)
+                $view = array_shift($view);
```

Comment:
Thanks for the input. Yes you're correct, the code should check if $allTicketView is empty and skip the loop.